### PR TITLE
ENG-1062 - Fix teacher dashboard to be able to display all the practice levels compactly

### PR DESCRIPTION
**Pull request by adamkecskes**
practice

## Summary by CodeRabbit

- **New Features**
	- Integrated the `moment` library for enhanced date manipulation and formatting.
	- Introduced new localization keys for improved support of practice levels.
	- Enhanced the teacher dashboard with better handling and display of practice levels for each content item.
	- Added new props and computed properties in the `progressDot` component for better tracking of student progress.

- **Bug Fixes**
	- Refined logic for fetching and processing game content, ensuring correct aggregation of practice levels.

- **Documentation**
	- Added Storybook configuration for the `PracticeLevel` component, allowing for better visualization and testing.

- **Chores**
	- Updated project dependencies to include the Ace Editor for improved code editing capabilities.

### CODE REVIEW

**Review by nwinter — APPROVED** [review]
Looks awesome!

File: package.json [hunk 1]
```diff
@@ -88,6 +88,7 @@
     "@stripe/stripe-js": "^1.32.0",
     "@usesuperflow/client": "^1.0.2",
     "@vimeo/player": "^2.17.1",
+    "ace-builds": "^1.36.0",
```
[the remark this answers is not in the document]
Comment (adamkecskes) [reply package.json:91]: @nick storybook threw an error because it was missing

[the remark this answers is not in the document]
Comment (nwinter) [reply package.json:91]: Should we put it in dev dependencies?

File: .storybook/preview.js [hunk 2]
```diff
@@ -182,6 +182,8 @@ window.me = {
 window._ = _
 window._.string = require('underscore.string')
 
+window.moment = require('moment')
```
Comment (mrfinch) [inline .storybook/preview.js:185]: could this be avoided? I guess it wouldn't change the bundle with moment already there but moment is hugeeee

File: ozaria/site/store/TeacherDashboardPanel.js [hunk 3]
```diff
@@ -12,20 +12,41 @@ export const DRAGGABLE_STATEMENT_COMPLETION = 'DRAGGABLE_STATEMENT_COMPLETION'
...
+  const level = new Level(content)
+
+  const solutionCode = level.getSolutions()
+    .filter((s) => s.language === language)
+    .find((s) => !s.testOnly)
+    ?.source || ''
+  const language = studentSessions[normalizedOriginal]?.codeLanguage || 'python'
+
+  const starterCode = level.getSampleCode()?.[language] || ''
+  const studentCode = studentSessions[normalizedOriginal]?.code?.['hero-placeholder']?.plan || ''
```
[the remark this answers is not in the document]
Comment (mrfinch) [reply ozaria/site/store/TeacherDashboardPanel.js:27]: I think studentCode appears but solution is blank that needs to be fixed

**Review by mrfinch — COMMENTED** [review]
can we add a check for NaN. I think it's a bug already not introduced by your PR.
Screenshot 2024-08-28 at 5 11 45 PM

**Pull request merged**